Index: kgb-arch-posix-by-slawek_1.0b4+ds/kgb_arch_posix_by_slawek.cpp
===================================================================
--- kgb-arch-posix-by-slawek_1.0b4+ds.orig/kgb_arch_posix_by_slawek.cpp
+++ kgb-arch-posix-by-slawek_1.0b4+ds/kgb_arch_posix_by_slawek.cpp
@@ -2178,9 +2178,9 @@ int _mode = 0;
   // Check arguments
   if (argc<2) {
       printf("KGB Archiver v1.0, (C) 2005-2006 Tomasz Pawlak\nBased on PAQ6 by Matt Mahoney\nmod by Slawek (poczta-sn@gazeta.pl)\n\n"
-      "Compression:\t\tkgb_arch.exe -<m> archive.kgb files <@list_files>\n"
-      "Decompression:\t\tkgb_arch.exe archive.kgb\n"
-      "Table of contests:\tmore < archive.kgb\n\n"
+      "Compression:\t\tkgb -<m> archive.kgb files <@files_list>\n"
+      "Decompression:\t\tkgb archive.kgb\n"
+      "Table of contents:\tmore < archive.kgb\n\n"
       "m argument\tmemory usage\n"
       "----------\t------------------------------\n"
       " -0       \t 2 MB (the fastest compression)\n"
